File: check-if-digits-are-equal-in-string-after-operations-i.ts
{
    function hasSameDigits(s: string): boolean {
        
        while (s.length !== 2) {
            let r = "";

            for (let i = 1; i < s.length; i++) {
                const sum = Number(s[i]) + Number(s[i - 1]);
    
                r += sum % 10;
            }
            s = r;
        }

        return s[0] == s[1]
    };

    console.log(hasSameDigits("3902"))
}

File: check-if-digits-are-equal-in-string-after-operations-i.rs
pub fn has_same_digits(s: String) -> bool {
    let s: Vec<u32> = s
        .chars()
        .map(|c| c.to_digit(10).unwrap())
        .collect::<Vec<_>>();

    let digits = (0..(s.len() - 2)).into_iter().fold(s, |acc, _| {
        acc.windows(2)
            .into_iter()
            .map(|window| (window[0] + window[1]) % 10)
            .collect::<Vec<_>>()
    });

    digits[0] == digits[1]
}
